dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/event/JdbcEventChangeLogStore.java

```diff
@@ -0,0 +1,92 @@
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.hisp.dhis.tracker.export.event;
+
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.hisp.dhis.program.UserInfoSnapshot;
+import org.hisp.dhis.tracker.export.event.EventChangeLog.Change;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.stereotype.Repository;
+
+@Repository("org.hisp.dhis.tracker.export.event.ChangeLogStore")
+@RequiredArgsConstructor
+class JdbcEventChangeLogStore {
+
+  private final JdbcTemplate jdbcTemplate;
+
+  private static final RowMapper<EventChangeLog> customEventChangeLogRowMapper =
+      (rs, rowNum) -> {
+        UserInfoSnapshot user = new UserInfoSnapshot();
+        user.setUsername(rs.getString("userName"));
+        user.setFirstName(rs.getString("firstname"));
+        user.setSurname(rs.getString("surname"));
+        user.setUid(rs.getString("useruid"));
+
+        return new EventChangeLog(
+            user,
+            rs.getTimestamp("updatedAt"),
+            new Change(
+                new EventChangeLog.DataValueChange(
+                    rs.getString("dataElementUid"),
+                    rs.getString("previousValue"),
+                    rs.getString("currentValue"))));
+      };
+
+  public List<EventChangeLog> getEventChangeLog(String eventUid) {
+    final String sql =
+        """
+            select
+              case
+                when cl.audittype = 'CREATE' then cl.previouschangelogvalue
+                when cl.audittype = 'UPDATE' and cl.currentchangelogvalue is null then cl.currentValue
+                when cl.audittype = 'UPDATE' and cl.currentchangelogvalue is not null then cl.currentchangelogvalue
+              end as currentValue,
+              case
+                when cl.audittype = 'DELETE' then cl.previouschangelogvalue
+                when cl.audittype = 'UPDATE' and cl.currentchangelogvalue is null then cl.previouschangelogvalue
+                when cl.audittype = 'UPDATE' and cl.currentchangelogvalue is not null then cl.previouschangelogvalue
+              end as previousValue, cl.created as updatedAt, cl.modifiedby as userName, cl.dataElementUid as dataElementUid, cl.firstname, cl.surname, cl.username, cl.useruid
+            from
+              (select t.created, d.uid as dataElementUid, t.modifiedby, t.audittype, u.firstname, u.surname, u.username, u.uid as useruid,
+                  LAG (t.value) OVER (PARTITION BY t.eventid, t.dataelementid ORDER BY t.created DESC) AS currentchangelogvalue,
+                  t.value as previouschangelogvalue,
+                  e.eventdatavalues -> d.uid  ->> 'value' as currentValue
+              from trackedentitydatavalueaudit t
+              join event e using (eventid)
+              join dataelement d using (dataelementid)
+              join userinfo u on u.username = t.modifiedby
```

Comment:
@muilpp I think the `trackedentitydatavalueaudit` table needs to change now and get a proper name like `trackedentitydatavaluechangelog`. This was agreed upon here https://dhis2.atlassian.net/browse/TECH-1659
so I think we should start doing these migrations

Reply:
The thing is this table needs a lot more changes than that. 
We should change the model and have previous value and current value in the same entry.
Then, we only store the current event data value in case it's created, if it's updated or deleted we store the previous only, so that should change too, but it also requires a migration.
Since we are tight on time, I think it's better to do these refactors/migrations after the release, so we can have a good plan on how to tackle all of this, Do you agree @lucaCambi77 @teleivo?

Reply:
I agree. We need more time to investigate all the consequences and what the ideal would be. I would therefore postpone any migrations of changelog related tables.

Reply:
ok, sorry, I haven't followed that part. We should also get rid of the `ChangeLogType.READ` type and move that to the real audit i think or remove it completely. Currently we add to change log in `addTrackedEntityChangeLog` of `DefaultTrackedEntityService` for read operations

Reply:
we will, we just don't have the time right now. soft freeze in super close